### PR TITLE
Implement Normalizeaddress hint

### LINKED ISSUE #568: `NormalizeAddress` hint
https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/common/storage.cairo#L14

https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/0.10.3.json#L53

### PR DESCRIPTION
**Pull request by har777**
Fix #568

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_others.go [hunk 1]
```diff
@@ -646,3 +647,66 @@ func createNondetElementsOverXHinter(resolver hintReferenceResolver, x uint64) (
...
+			//> assert (2**250 < ADDR_BOUND <= 2**251) and (2 * 2**250 < PRIME) and (
+			//>         ADDR_BOUND * 2 > PRIME), \
+			//>    'normalize_address() cannot be used with the current constants.'
+			//> ids.is_small = 1 if ids.addr < ADDR_BOUND else 0
+
+			// ADDR_BOUND and PRIME are constants: https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/common/storage.cairo#L6
+			// so the assert check is skipped as the constants satisfy them
+
+			// 2 ** 251 -256
+			// TODO: directly set the felt limbs to save time?
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_others.go:672]: this sound like a good idea for sure

**Comment by MaksymMalicki** [comment]
The code looks good. Do you know why the hint code is not detected? If it is just simple typo to fix, the I'm giving approve

**Comment by har777** [comment]
> The code looks good. Do you know why the hint code is not detected? If it is just simple typo to fix, the I'm giving approve

Ah that integration test needs another hint assigned to someone else. I'll move it to the in_progress folder.

File: pkg/hintrunner/zero/zerohint_others.go (hunk 1 above)
Comment (har777) [reply pkg/hintrunner/zero/zerohint_others.go]: done

File: pkg/hintrunner/zero/zerohint_others.go [hunk 2]
```diff
@@ -694,3 +694,60 @@ func createNondetElementsOverXHinter(resolver hintReferenceResolver, x uint64) (
...
+//
+// `newNormalizeAddressHint` takes 2 arguments
+//   - `isSmall` represents the address where the result of the comparison is stored
+//   - `addr` represents the address whose value is checked against ADDR_BOUND
+func newNormalizeAddressHint(isSmall, addr hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "NormalizeAddress",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> # Verify the assumptions on the relationship between 2**250, ADDR_BOUND and PRIME.
+			//> ADDR_BOUND = ids.ADDR_BOUND % PRIME
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_others.go:709]: what would be the sense of this modulo here? making sure this hint is not utilised in a context with a prime too big (or too small)? but i think `PRIME` is also a constant no? anyway... goddamn python vm

File: pkg/hintrunner/zero/zerohint_others_test.go [hunk 3]
```diff
@@ -604,5 +604,33 @@ func TestZeroHintOthers(t *testing.T) {
...
+					return newNormalizeAddressHint(
+						ctx.operanders["is_small"],
+						ctx.operanders["addr"],
+					)
+				},
+				check: varValueEquals("is_small", feltUint64(0)),
+			},
+			{
+				operanders: []*hintOperander{
+					{Name: "addr", Kind: apRelative, Value: feltString("3618502788666131106986593281521497120414687020801267626233049500247285300991")},
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_others_test.go:623]: this is `2 ** 251 - 255` ?

File: pkg/hintrunner/zero/zerohint_others_test.go [hunk 4]
```diff
@@ -604,5 +604,33 @@ func TestZeroHintOthers(t *testing.T) {
 				check: apValueEquals(feltUint64(0)),
 			},
 		},
+		"NormalizeAddress": {
+			{
+				operanders: []*hintOperander{
+					{Name: "addr", Kind: apRelative, Value: feltString("3618502788666131106986593281521497120414687020801267626233049500247285300992")},
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_others_test.go:610]: this is `2 ** 251 - 256` ?

**Review by TAdev0 — APPROVED** [review]
left comments but lgtm

File: pkg/hintrunner/zero/zerohint_others_test.go (hunk 4 above)
Comment (har777) [reply pkg/hintrunner/zero/zerohint_others_test.go:610]: `2 ** 251 - 256`

Comment (har777) [reply pkg/hintrunner/zero/zerohint_others_test.go:610]: will add a comment

File: pkg/hintrunner/zero/zerohint_others_test.go (hunk 3 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_others_test.go:625]: yes perfect

**Pull request merged**